specs/interop/shared-lockbox.md

```diff
@@ -0,0 +1,340 @@
+
+- Only `SuperchainConfig` address MUST be allowed to interact.
+- The function MUST add the specified address to the mapping of authorized portals.
+- The function MUST emit the `AuthorizedPortal` event when a portal is successfully added.
+
+```solidity
+function authorizePortal(address _portal) external;
+```
+
+**`ETHLocked`**
+
+MUST trigger when `lockETH` is called
+
+```solidity
+event ETHLocked(address indexed portal, uint256 amount);
+```
+
+**`ETHUnlocked`**
+
+MUST trigger when `unlockETH` is called
+
+```solidity
+event ETHUnlocked(address indexed portal, uint256 amount);
+```
+
+**`AuthorizedPortal`**
+
+MUST trigger when `authorizePortal` is called
+
+```solidity
+event AuthorizedPortal(address indexed portal);
+```
+
+#### `LiquidityMigrator`
+
+This contract is meant to be used as intermediate step for the liquidity migration.
+It main purpose is to transfer the whole ETH balance from the `OptimismPortal` to the `SharedLockbox`.
+
+**Interface and properties:**
+
+**`migrateETH`**
+
+Transfers the entire ETH balance from the OptimismPortal to the SharedLockbox.
+
+- It MUST transfer the whole ETH balance to the `SharedLockbox` when called.
+
+```solidity
+function migrateETH() external;
+```
+
+### Contracts upgrades
+
+#### `SuperchainConfig`
+
+This contract will be updated to manage and keep track the dependency graph.
+It will be queried as the source of truth to get which chains are part of the Superchain.
+It assumes the role of a `dependencyManager` per `SystemConfig` contract involved.
+It will also allow to add a chain to the op-governed cluster and update each chain’s dependency set.
+
+**Interface and properties**
+
+The `SuperchainConfig` contract will add the following storage layout and function:
+
+**`SHARED_LOCKBOX`**
+
+- An immutable address pointing to the `SharedLockbox` contract.
+- This address MUST immutable because there's only one `SharedLockbox` for each cluster.
+
+**`systemConfigs`**
+
+- A mapping that associates chain IDs with their respective SystemConfig addresses.
+- This will be used when updating dependencies along each chain.
+
+**`dependencySet`**
+
+- An `EnumerableSet` that stores the current list of chain IDs in the dependency set.
+- This MUST replicate the same state as the one stored in the `L1BlockInterop` on L2 for each chain.
+
+**`addChain`**
+
+The `addChain` function adds a new chain to the op-governed cluster.
+
+- The function MUST only be callable by the authorized `updater` role of the `SuperchainConfig`.
+- The function MUST NOT add a chain ID to the dependency set if it is already included.
+- The function MUST update dependencies to form a complete mesh graph,
+  linking the new chain with all existing chains in the set.
+- The function MUST store the provided `SystemConfig` address in the systemConfigs mapping.
+- The function MUST whitelist the new chain's portal in the `SharedLockbox`.
+- The function MUST emit the `ChainAdded` event.
```

Comment:
with what arguments? I know it's seen in the example impl, but it's not formally defined if i didn't miss it, like other events

Reply:
good catch, i missed it